src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,10 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(e -> e % 2 != 0 ? numbers.get(e) - 1 : numbers.get(e))
+                .filter(i -> i % 2 != 0)
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
* new line
* `getAsDouble()`  will throw that exception for you
```suggestion
                .average()
                .getAsDouble()
```